File: 待办清单功能/ruankai2/app/src/main/java/com/byted/camp/todolist/beans/State.java
package com.byted.camp.todolist.beans;


public enum State {
    TODO(0), DONE(1);

    public final int intValue;

    State(int intValue) {
        this.intValue = intValue;
    }

    public static State from(int intValue) {
        for (State state : State.values()) {
            if (state.intValue == intValue) {
                return state;
            }
        }
        return TODO; // default
    }

    public static int refrom(State state) {
        if(state==State.DONE)
            return 1;
        return 0;
    }
}

File: 12.25整合代码/MyApplication2/app/src/main/java/com/example/myapplication/todolist/beans/State.kt
package com.example.myapplication.todolist.beans

enum class State(val intValue: Int) {
    TODO(0), DONE(1);

    companion object {
        fun from(intValue: Int): State {
            for (state in com.example.myapplication.todolist.beans.State.values()) {
                if (state.intValue == intValue) {
                    return state
                }
            }
            return TODO // default
        }

        fun refrom(state: State): Int {
            return if (state == DONE) 1 else 0
        }
    }
}
